jerry-core/ecma/builtin-objects/ecma-builtin-bigint.c

```diff
@@ -36,6 +53,353 @@
  * @{
  */
 
+/**
+ * The BigInt object's 'asIntN' and 'asUintN' routines
+ *
+ * See also:
+ *          ECMA-262 v5, 11.0
+ *
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value.
+ */
+static ecma_value_t
+ecma_builtin_bigint_object_as_int_n (ecma_value_t this_arg, /**< 'this' argument */
+                                     ecma_value_t bits, /**< number of bits */
+                                     ecma_value_t bigint, /**< bigint number */
+                                     bool is_signed) /**< The operation is signed */
+{
+  JERRY_UNUSED (this_arg);
+
+  ecma_number_t input_bits;
+  ecma_value_t bit_value = ecma_op_to_index (bits, &input_bits);
+
+  if (ECMA_IS_VALUE_ERROR (bit_value))
+  {
+    return bit_value;
+  }
+
+  ecma_value_t bigint_value = ecma_bigint_to_bigint (bigint, false);
+
+  if (ECMA_IS_VALUE_ERROR (bigint_value))
+  {
+    return bigint_value;
+  }
+
+  if (input_bits == 0 || bigint_value == ECMA_BIGINT_ZERO)
+  {
+    ecma_free_value (bigint_value);
+    return ECMA_BIGINT_ZERO;
+  }
+
+  const uint32_t size_of_divisor_in_bits = (sizeof (uint32_t) * sizeof (uint64_t));
+
+  uint64_t whole_part = (uint64_t) floor (input_bits / size_of_divisor_in_bits);
+  uint32_t remainder = (uint32_t) fmod (input_bits, size_of_divisor_in_bits);
```

Comment:
The input_bits should be an integer number. Furthermore JerryScript has a maximum length for bigints, any number greater than that does not need any clamping (just increase its refcount and return with it).

Reply:
The head of the code is coming from the standard. So the input_bits should be a number according to the ToIndex method.
(https://tc39.es/ecma262/#sec-bigint.asintn)